vault_core/src/vault.rs

```diff
@@ -2079,4 +2173,52 @@ mod tests {
         let mut vault = make_test_vault(0, 0, 1000, 1000, DelegationState::default());
         assert_eq!(vault.burn_with_fee(0, 0), Err(VaultError::VaultBurnZero));
     }
+
+    #[test]
+    fn test_calculate_min_amount_out() {
+        let vault = make_test_vault(0, 0, 1000, 1000, DelegationState::default());
+        let amount_out = 100;
+        let min_amount_out = vault
+            .calculate_min_supported_mint_out(amount_out, 100)
+            .unwrap();
+        assert_eq!(min_amount_out, 99);
+    }
+
+    #[test]
+    fn test_calculate_min_amount_out_with_withdrawal_fee() {
+        let vault = make_test_vault(0, 100, 1000, 1000, DelegationState::default());
+        let amount_out = 100;
+        let min_amount_out = vault
+            .calculate_min_supported_mint_out(amount_out, 100)
+            .unwrap();
+        assert_eq!(min_amount_out, 98);
+    }
+
+    #[test]
+    fn test_calculate_min_amount_out_slippage_too_low() {
+        let vault = make_test_vault(0, 100, 1000, 1000, DelegationState::default());
+        let amount_out = 100;
+        assert!(vault
+            .calculate_min_supported_mint_out(amount_out, Vault::MIN_WITHDRAWAL_SLIPPAGE_BPS - 1)
+            .is_err());
+    }
+
+    #[test]
+    fn test_check_min_amount_out_ok() {
+        let vault = make_test_vault(0, 100, 1000, 1000, DelegationState::default());
+        let amount_out = 100;
+        let min_amount_out = vault
+            .calculate_min_supported_mint_out(amount_out, 50)
+            .unwrap();
+        assert!(vault
+            .check_min_supported_mint_out(amount_out, min_amount_out)
+            .is_ok());
+    }
+
+    #[test]
+    fn test_check_min_amount_out_too_high() {
+        let vault = make_test_vault(0, 100, 1000, 1000, DelegationState::default());
+        let amount_out = 100;
+        assert!(vault.check_min_supported_mint_out(amount_out, 99).is_err());
```

Comment:
nit: `assert_eq!` with the expected error type

e.g. `assert_eq!(vault.burn_with_fee(0, 0), Err(VaultError::VaultBurnZero));`